### PR TITLE
EXSWHTEC-115 - Implement tests for hipLaunchCooperativeKernel APIs

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
- Negative parameter tests for hipLaunchCooperativeKernel and hipLaunchCooperativeKernelMultiDevice
- Positive parameter tests for hipLaunchCooperativeKernel
- Basic positive scenarios for hipLaunchCooperativeKernel and hipLaunchCooperativeKernelMultiDevice
- Negative scenario for hipLaunchCooperativeMultiDevice with multiple kernels on the same device.

Depends-on: #58

### CODE REVIEW

File: catch/unit/executionControl/hipFuncGetAttributes.cc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    const auto minor = GetDeviceAttribute(hipDeviceAttributeComputeCapabilityMinor, 0);
+    REQUIRE(attr.binaryVersion == major * 10 + minor);
+#elif HT_AMD
+    REQUIRE(attr.binaryVersion > 0);
+#endif
+  }
+
+  SECTION("cacheModeCA") { REQUIRE((attr.cacheModeCA == 0 || attr.cacheModeCA == 1)); }
+
+#if HT_NVIDIA // Disabled on AMD due to defect - EXSWHTEC-242
```
Comment (chrispaquot) [inline catch/unit/executionControl/hipFuncGetAttributes.cc]: just disable the test case.

File: catch/unit/executionControl/hipFuncGetAttributes.cc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            GetDeviceAttribute(hipDeviceAttributeMaxSharedMemoryPerBlock, 0));
+  }
+}
+
+TEST_CASE("Unit_hipFuncGetAttributes_Negative_Parameters") {
+  SECTION("attr == nullptr") {
+    HIP_CHECK_ERROR(hipFuncGetAttributes(nullptr, reinterpret_cast<void*>(attribute_test_kernel)),
+                    hipErrorInvalidValue);
+  }
+#if HT_NVIDIA // Disabled on AMD due to defect - EXSWHTEC-241
```
Comment (chrispaquot) [inline catch/unit/executionControl/hipFuncGetAttributes.cc]: same here.

**Review by chrispaquot — APPROVED** [review]

**Comment by rorake** [comment]
@mirza-halilcevic 
Will you please resolve the merge conflicts for this PR?

**Comment by rakesroy** [comment]
PR has been merged into develop branch via commit 380ace735f1a373a90690a410de8164f6d962e5d.

**Pull request closed without merging**